packages-native/intro-screen/src/SwipeableContainer.tsx

```diff
@@ -0,0 +1,310 @@
+        const isLastSlide = activeIndex === props.slides.length - 1;
+        const isFirstSlide = activeIndex === 0;
+
+        const skipBtn = (!isFirstSlide && _renderPrevButton(props)) || (!isLastSlide && _renderSkipButton(props));
+        const btn = isLastSlide ? _renderDoneButton(props) : _renderNextButton(props);
+
+        return (
+            <View style={[styles.paginationContainer, props.styles.paginationContainer]}>
+                <View style={[styles.paginationDots, props.styles.paginationDots]}>
+                    {!props.hidePagination &&
+                        props.slides.length > 1 &&
+                        props.slides.map((_, i) => (
+                            <Touchable
+                                key={i}
+                                style={[
+                                    styles.dot,
+                                    _rtlSafeIndex(i) === activeIndex
+                                        ? props.styles.activeDotStyle
+                                        : props.styles.dotStyle
+                                ]}
+                                onPress={() => _onPaginationPress(i)}
+                            />
+                        ))}
+                </View>
+                {!props.bottomButton && (
+                    <Fragment>
+                        {btn}
+                        {skipBtn}
+                    </Fragment>
+                )}
+                {props.bottomButton && (
+                    <View style={styles.bottomButtonsContainer}>
+                        {props.numberOfButtons === 2 && skipBtn}
+                        {btn}
+                    </View>
+                )}
+            </View>
+        );
+    };
+
+    const _rtlSafeIndex = (i: number): number => (isAndroidRTL ? props.slides.length - 1 - i : i);
+
+    const _onMomentumScrollEnd = useCallback(
+        (e: NativeSyntheticEvent<any>) => {
+            const offset = e.nativeEvent.contentOffset.x;
+            const newIndex = _rtlSafeIndex(Math.round(offset / width));
+            if (newIndex === activeIndex) {
+                return;
+            }
+            const lastIndex = activeIndex;
+            setActiveIndex(newIndex);
+            if (props.onSlideChange) {
+                props.onSlideChange(newIndex, lastIndex);
+            }
+        },
+        [activeIndex, width]
+    );
+
+    /**
+     * Readjust the size of the slides if the size is different than the device dimensions
+     */
+    const _onLayout = useCallback(
+        (e: LayoutChangeEvent) => {
+            const newWidth = e.nativeEvent.layout.width;
+            if (newWidth !== width) {
+                setWidth(newWidth);
+            }
+        },
+        [width]
+    );
+
+    // eslint-disable-next-line no-unused-vars,@typescript-eslint/no-unused-vars
+    const { hidePagination, skipLabel, doneLabel, nextLabel, prevLabel, ...otherProps } = props;
+
+    const _setRef = useCallback((ref: any) => {
```

Comment:
Can this use https://reactjs.org/docs/hooks-reference.html#useref instead?

Reply:
Done, actually is a better option than useState in this case